include/aws/http/status_code.h

```diff
@@ -0,0 +1,107 @@
+#ifndef AWS_HTTP_STATUS_CODE_H
+#define AWS_HTTP_STATUS_CODE_H
+
+/*
+ * Copyright 2010-2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+/*
+ * Define most of the http response codes we probably will use.
+ * https://en.wikipedia.org/wiki/List_of_HTTP_status_codes
+ */
+enum http_status_code {
```

Comment:
There's already a status-code enum defined in `private/http_impl.h`
```
/**
 * Status codes that affect internal processing.
 * This is NOT a definitive list of codes.
 */
enum aws_http_status {
    AWS_HTTP_STATUS_UNKNOWN = -1, /* Invalid status code. Not using 0 because it's technically a legal value */
    AWS_HTTP_STATUS_100_CONTINUE = 100,
    AWS_HTTP_STATUS_101_SWITCHING_PROTOCOLS = 101,
    AWS_HTTP_STATUS_204_NO_CONTENT = 204,
    AWS_HTTP_STATUS_304_NOT_MODIFIED = 304,
};
```
Once upon a time this was a public enum, but I made it private to prevent accidentally building an API where users are required to use the enums, since any numeric value < 1000 is technically allowed. I don't have a huge issue with making it public again, just explaining why it was hidden away.

Couple of points:
1) I'd rather stick to the [official](https://www.iana.org/assignments/http-status-codes/http-status-codes.txt) status codes, rather than this wikipedia list which includes stuff like codes specific to Shopify
2) maybe it's just me? but I liked including the number value in the name of the enum values. `AWS_HTTP_STATUS_100_CONTINUE` vs `AWS_HTTP_STATUS_CONTINUE`?

Reply:
I like the second idea. I had the hesitation when changing all the numbers to names, which makes the numbers implicit.